Source/WebGPU/WGSL/Overload.cpp

```diff
@@ -363,10 +362,83 @@ bool OverloadResolver::unify(const AbstractValue& parameter, unsigned argumentVa
     return *resolvedValue == argumentValue;
 }
 
-void OverloadResolver::assign(TypeVariable variable, Type* type, Substitution::Inferred inferred)
+bool OverloadResolver::assign(TypeVariable variable, Type* type, Substitution::Inferred inferred)
 {
     logLn("assign ", variable, " => ", *type);
+    if (variable.constraints) {
+        auto* primitive = std::get_if<Types::Primitive>(type);
+        if (!primitive)
+            return false;
+
+        switch (primitive->kind) {
+        case Types::Primitive::AbstractInt:
```

Comment:
Maybe I'm not understanding it correctly, but this seems backwards to me. I'd expect the switch to switch over the constraint, and then for each constraint kind, ask if the primitive fits the constraint. Here, it looks like you're switching over the primitive type and then asking which constraints would fit the primitive. Is this intentional?